ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/sse/SSEConfig.kt

```diff
@@ -13,6 +13,7 @@ import kotlin.time.Duration.Companion.milliseconds
 public class SSEConfig {
     internal var showCommentEvents = false
     internal var showRetryEvents = false
+    internal var deserialize: (String) -> Any = { s -> s }
```

Comment:
Looks like we're missing the typeInfo here required to solve the general use-case for deserializing different types.  I think it may need to be `(TypeInfo) -> (String) -> Any` so when using a `StringFormat` like `Json`, we can supply it like `deserialize = { typeInfo -> Json.serializersModule.serializer(typeInfo).let { serializer -> {{ Json.decodeFromString(serializer, it) }} } }`.  Rather ugly, but we could provide some helpers somewhere.

Reply:
From the API side, this will need to use `reified T` and `typeOf<T>()` to pass to the session.